DataFormats/ParticleFlowReco/interface/PFRecHitSoA.h

```diff
@@ -0,0 +1,33 @@
+#ifndef DataFormats_ParticleFlowReco_interface_PFRecHitSoA_h
+#define DataFormats_ParticleFlowReco_interface_PFRecHitSoA_h
+
+#include <Eigen/Core>
+#include <Eigen/Dense>
+
+#include "DataFormats/ParticleFlowReco/interface/PFLayer.h"
+#include "DataFormats/SoATemplate/interface/SoACommon.h"
+#include "DataFormats/SoATemplate/interface/SoALayout.h"
+#include "DataFormats/SoATemplate/interface/SoAView.h"
+
+namespace reco {
+
+  using PFRecHitsNeighbours = Eigen::Matrix<int32_t, 8, 1>;
+  GENERATE_SOA_LAYOUT(PFRecHitSoALayout,
+                      SOA_COLUMN(uint32_t, detId),
+                      SOA_COLUMN(float, energy),
+                      SOA_COLUMN(float, time),
+                      SOA_COLUMN(int, depth),
+                      SOA_COLUMN(PFLayer::Layer, layer),
+                      SOA_EIGEN_COLUMN(PFRecHitsNeighbours,
+                                       neighbours),  // Neighbour indices (or -1); order: N, S, E, W, NE, SW, SE, NW
```

Comment:
Maybe moving the comment to preceding line allows getting better formatting?

Reply:
If different GPU threads are going to access different neighbours of the same SoA entry, then this should probably be an `edm::StdArray`.

If different GPU threads are going to access the same neighbour of different SoA entries, then this should probably stay an Eigen::Matrix.